src/builtins/core/date.rs

```diff
@@ -235,20 +231,14 @@ impl PlainDate {
         //    duration.[[Nanoseconds]]).
         // 6. Let days be duration.[[Days]] + BalanceTimeDuration(norm,
         //    "day").[[Days]].
-        let days = duration.days().checked_add(
-            &TimeDuration::from_normalized(duration.time().to_normalized(), Unit::Day)?.0,
-        )?;
+        let days = duration.days().saturating_add(
```

Comment:
`saturating_add` actually has me a little concerned looking back as it may hide some nasty bugs in the future. I'm going to add a method to address this.